src/pages/RegisterPage.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  VStack,
+  Stack,
+  Heading,
+  Text,
+  useToast,
+  useColorModeValue,
+  InputGroup,
+  InputRightElement,
+} from '@chakra-ui/react';
+import { ViewIcon, ViewOffIcon } from '@chakra-ui/icons';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { Link as RouterLink } from 'react-router-dom';
+import { Link as ChakraLink } from '@chakra-ui/react';
+import { FormFields } from '../types/types';
+import CustomToast from '../components/CustomToast';
+import { useAppDispatch } from '../store/hooks';
+import { loginUser } from '../features/user/userSlice';
+import { registerCustomer } from '../services';
+
+const RegisterPage: React.FC = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const dispatch = useAppDispatch();
+  const toast = useToast();
+
+  const [form, setForm] = useState<FormFields>({
+    email: '',
+    password: '',
+    firstName: '',
+    lastName: '',
+    confirmPassword: '',
+  });
+
+  const [errors, setErrors] = useState<Partial<FormFields>>({});
+  const [error, setError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const nameRegex = /^[A-Za-zА-Яа-яёЁ\s-]+$/;
```

Comment:
would better move out this constatn and import here (or somewhere else if needed)